_posts/2020-08-11-backports-application.adoc

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: From zero to hero on backporting pull-requests
+date: 2020-08-11
+tags: user-story
+author: gastaldi
+---
+= From zero to hero on backporting pull-requests
+
+Before we dig into the solution, let me give you a quick explanation about our release process
+
+== Release process
+=== Minor versions
+Our master branch is always ready for the next minor release. That means that any pull-requests merged in there are available in the next minor release. This process is usually very smooth and involves no backporting.
+
+=== Patch releases or CRs
+Whenever a new patched version (or a CR) is about to be released in Quarkus, our Release engineer starts backporting commits from pull-requests that were merged in the master branch.
+How do they know which pull-request to grab? We have a `triage/backport?` label that our team add to pull-requests for features that would be worth having it in the upcoming release.
+
+== What the app does?
+
+The application basically queries a GitHub repository's merged pull requests and closed issues (using the GraphQL API) containing a certain label and changes their milestones to one that is selected in the UI (and removing the certain label afterwards)
```

Comment:
I think it's a bit limiting.

We need to explain:
- why we needed it: the number of pull requests to backport is enormous and doing it via the GitHub UI is cumbersome and takes a lot of time (click PR, copy paste commit hashes to cherry-pick, remove label, affect milestone, affect milestone to all fixes issues, next) + some limitation of the UI (it's not possible to sort by merge date to avoid conflicts)
- it also simplifies the cherry picking